text/0000-innerhtml-binding.md

```diff
@@ -0,0 +1,108 @@
+- Start Date: 2019-10-03
+- RFC PR: (leave this empty)
+- Lightning Web Component Issue: (leave this empty)
+
+# Summary
+
+Applications dealing with rich content, like commerce apps, need to render 
+this content as raw HTML, typically coming from a database or a CMS. Today, 
+the data binding capability always escapes the HTML for security reasons,
+which makes the rich content data not rendered as expected.
+
+# Basic example
+
+The following template render some HTML content as innerHTML  
+
+```
+function createMarkup() { return {__html: 'First &middot; Second'}; };
+
+<template>
+    <div lwc:innerHTML={createMarkup()}/>
+</template>
+```
+
+# Motivation
+
+It is common in customer facing apps to render rich content coming from a
+CMS or equivalent.  
+
+As of today, and according to the LWC documentation, this capability is 
+supported by acting directly with the DOM within `renderedCallback()`.
+The templating mechanism does not provide any binding capability that renders
+some content as raw HTNL.  
+
+This breaks with SSR, as `renderedCallback()` and DOM manipulation won't be
+allowed when a component is rendered on the server. Moreover, even if it was
+possible, it would make the client side hydration more complex as the content
+is not generated as part of the VDOM.  
+
+Having to act on the DOM is also not user friendly compared to a binding
+facility.  
+
+
+# Detailed design
+
+It seems that there is a consensus among the most popular libraries to expose the
+innerHTML property of an element, although the exposed name for this property varies:
+
+React: [https://zhenyong.github.io/react/tips/dangerously-set-inner-html.html]().  
+Angular: [https://angular.io/guide/template-syntax#property-binding-or-interpolation]().  
+VueJS: [https://vuejs.org/v2/guide/syntax.html#Raw-HTML]().  
+
+The Angular choice of `innerHTML` feels the most natural, while the React `dangerouslyInnerHTML`
+has probably be named this way to not conflict with any potential component property.
+To avoid any name collision, LWC can prefix it with the technical "lwc:" prefix.
+
+React makes it even more secure by forcing the HTML to be added to a temporary 
+object with a `__html` property. This avoids undesired assignment from a method

```

Comment:
I am not convinced about the benefit of adding this `__html` property on an object. The `lwc` prefix is fairly uncommon in the template, adding this extra layer of "security" is feels unnecessary.